### PR TITLE
chore: replace modal hooks with UIKit modals in model and model version pages

### PR DESCRIPTION
**Pull request by keita-determined**
## Description
[WEB-1103]

## Test Plan
- Check if all modals from model and model version work the same as before
- Check if tags and metadata validations work in `create model modal`
  - No duplicate keys or tags
  - No empty key or tags

## Commentary (optional)
- The behavior of `create model modal` in checkpoint modal is changed due to the difficulty of migration with checkpoint modal
  - It used to open checkpoint modal again after creating model, but it doesn't with new `create model modal` since we don't use context holder anymore.  I think this should be able to fix it in checkpoint modal migration. I added `TODO` comment in each checkpoint modal usage

## Checklist

- [x] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

[WEB-1103]: https://hpe-aiatscale.atlassian.net/browse/WEB-1103?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webui/react/src/components/ModelVersionDeleteModal.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  };
+
+  return (
+    <Modal
+      danger
+      size="small"
+      submit={{ handler: handleOk, text: 'Delete Version' }}
+      title="Confirm Delete Model Version">
+      <div>
+        Are you sure you want to delete &quot; V{modelVersion.version}&quot; from this model?
```
Comment (mapmeld) [inline webui/react/src/components/ModelVersionDeleteModal.tsx:41]: For deleting model version, let's write it `"Version ${modelVersion.version}"` for consistency with past modal.  If I give the model version a custom name, then "Version 1" and the name are more prominent than "V1".

Screen Shot 2023-03-30 at 10 24 37 AM

File: webui/react/src/hooks/useModal/Checkpoint/useCheckpointFlow.tsx [hunk 2]
```diff
@@ -22,12 +23,14 @@ export const useCheckpointFlow = ({
...
   title: string;
 }): Return => {
+  const modelCreateModal = useModal(ModelCreateModal);
   const {
     contextHolder: modalCheckpointRegisterContextHolder,
     modalOpen: openModalCheckpointRegister,
   } = useModalCheckpointRegister({
     onClose: (reason?: ModalCloseReason, checkpoints?: string[]) => {
-      if (checkpoints) openModalCreateModel({ checkpoints });
+      // TODO: fix the behavior along with checkpoint modal migration
```
Comment (mapmeld) [inline webui/react/src/hooks/useModal/Checkpoint/useCheckpointFlow.tsx:32]: same comment as on useCheckpointFlow re: what needs to be fixed here in the future

File: webui/react/src/components/ModelCreateModal.tsx [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+    }
+  };
+
+  const onCloseModal = () => {
+    onClose?.();
+  };
+
+  const onOpenDetails = () => {
+    setIsDetailExpanded(true);
+    form.resetFields();
```
[the remark this answers is not in the document]
Comment (keita-determined) [reply webui/react/src/components/ModelCreateModal.tsx:107]: nice catch!

**Review by mapmeld — APPROVED** [review]
LGTM, thanks for updates

**Pull request merged**